Makefile

```diff
@@ -54,86 +88,99 @@ default: release
-		-A clippy::needless_range_loop -D rust-2018-idioms -A clippy::redundant_closure \
-		-A clippy::match_wild_err_arm -A clippy::blacklisted_name
+## Development builds
+## ------------------
+
+all: format build test
 
 dev: format clippy
 	@env FAIL_POINT=1 make test
 
 build:
 	cargo build  --no-default-features --features "${ENABLE_FEATURES}"
 
-ctl:
-	cargo build --release --no-default-features --features "${ENABLE_FEATURES}" --bin tikv-ctl
-	@mkdir -p ${BIN_PATH}
-	@cp -f ${CARGO_TARGET_DIR}/release/tikv-ctl ${BIN_PATH}/
-
 run:
 	cargo run --no-default-features --features  "${ENABLE_FEATURES}" --bin tikv-server
 
+
+## Release builds (optimized dev builds)
+## ----------------------------
+
+# These builds are heavily optimized, but only use thinLTO, not full
+# LTO, and they don't include debuginfo by default.
+
 # An optimized build suitable for development and benchmarking, by default built
 # with RocksDB compiled with the "portable" option, for -march=x86-64 (an
 # sse2-level instruction set), but with sse4.2 and the PCLMUL instruction
 # enabled (the "sse" option)
 release:
-	make dist_release
+	cargo build --release --no-default-features --features "${ENABLE_FEATURES}"
 
 # An optimized build that builds an "unportable" RocksDB, which means it is
 # built with -march native. It again includes the "sse" option by default.
 unportable_release:
-	make dist_unportable_release
+	ROCKSDB_SYS_PORTABLE=0 make release
 
 # An optimized build with jemalloc memory profiling enabled.
 prof_release:
-	make dist_prof_release
+	ENABLE_FEATURES=mem-profiling make release
 
 # An optimized build instrumented with failpoints.
 # This is used for schrodinger chaos testing.
 fail_release:
-	make dist_fail_release
+	FAIL_POINT=1 make release
 
 # The target used by CI/CD to build the distributable release artifacts.
 # Individual developers should only need to use the `dist_` rules when working
 # on the CI/CD system.
 dist_release:
-	make build_release
+	make build_dist_release
 	@mkdir -p ${BIN_PATH}
 	@cp -f ${CARGO_TARGET_DIR}/release/tikv-ctl ${CARGO_TARGET_DIR}/release/tikv-server ${CARGO_TARGET_DIR}/release/tikv-importer ${BIN_PATH}/
 	bash scripts/check-sse4_2.sh
 
+
+## Distribution builds (true release builds)
+## -------------------
+
+# These builds are fully optimized, with LTO, and they contain
+# debuginfo. They take a very long time to build, so it is recommended
+# not to use them.
+
+# Build with release flag as if it were for distribution, but without
+# additional sanity checks and file movement.
+build_dist_release:
+	make x-build-dist
+
+# Temporary backwards compatibility
+build_release:
+	@echo "the build_release rule no longer exists. you might want 'build_dist_release' or 'release'"
+	@exit 1
+
 # Distributable bins with SSE4.2 optimizations
 dist_unportable_release:
-	ROCKSDB_SYS_PORTABLE=0 make release
+	ROCKSDB_SYS_PORTABLE=0 make dist_release
 
 # Distributable bins with jemalloc memory profiling
 dist_prof_release:
```

Comment:
I guess we can remove `dist_prof_release` and `dist_fail_release`, mostly we can't use them in production.

Reply:
Done.